collections/_posts/2019-08-24-adding-custom-doh-resolvers-to-dnscloak.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Adding Custom DNS over HTTPS Resolvers to DNSCloak
+author: nitrohorse
+layout: post
+tags:
+  - guides
+  - dns
+  - ios
+---
+
+**[DNSCloak](https://apps.apple.com/us/app/dnscloak-secure-dns-client/id1452162351)** is an [open-source](https://github.com/s-s/dnscloak) DNSCrypt and DNS over HTTPS (DoH) client for iOS; giving users the ability to encrypt their DNS requests. While highly configurable, its user interface can be unintuitive and doesn't easily allow users to add custom DoH resolvers apart from the default ["public-resolvers" list that DNSCrypt provides](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v2/public-resolvers.md).
+
+This guide will show how to use DNSCloak to connect to your favorite public resolver that supports DNSCrypt or DoH.
+
+### Adding a Custom Resolver
+
+You'll notice DNSCloak provides a "Config Editor":
+
+![](/assets/img/2019-08-24-dnscloak/config-editor.jpeg){: .w-50}
+
+We can learn about the various configuration options from the [example config file](https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-proxy/example-dnscrypt-proxy.toml) in dnscrypt-proxy's code repository. If you scroll all the way to the bottom, you'll find a `[static.'myserver']` section along with a `stamp` property. This stamp is for adding the [DNS stamp](https://dnscrypt.info/stamps-specifications) of your resolver.
+
+> Server stamps encode all the parameters required to connect to a secure DNS server as a single string. Think about stamps as QR code, but for DNS.
```

Comment:
```suggestion
According to the DNSCrypt project, DNS stamps encode all the parameters required to connect to an encrypted DNS server as a single string. You can think about stamps as QR code, but for DNS.
```

I would probably not directly quote (partially because I am now realizing the blog doesn't have block quote support yet) and I would move this sentence to the end of the prior paragraph rather than have it on a separate line.